rct229/rulesets/ashrae9012019/section22/section22rule12.py

```diff
@@ -0,0 +1,106 @@
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.rulesets.ashrae9012019.ruleset_functions.get_heat_rejection_loops_connected_to_baseline_systems import (
+    get_heat_rejection_loops_connected_to_baseline_systems,
+)
+from rct229.utils.assertions import getattr_
+from rct229.utils.jsonpath_utils import find_all
+
+
+class Section22Rule12(RuleDefinitionListIndexedBase):
+    """Rule 12 of ASHRAE 90.1-2019 Appendix G Section 22 (Chilled water loop)"""
+
+    def __init__(self):
+        super(Section22Rule12, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section22Rule12.HeatRejectionRule(),
+            index_rmr="baseline",
+            id="22-12",
+            description="he heat rejection system shall be a single loop, modeled with a single cooling tower.",
+            ruleset_section_title="HVAC - Chiller",
+            standard_section="Section 22 CHW&CW Loop",
+            is_primary_rule=True,
+            rmr_context="ruleset_model_descriptions/0",
+            list_path="$.heat_rejections[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmd_b = context.baseline
+        heat_rejection_loop_ids_b = (
+            get_heat_rejection_loops_connected_to_baseline_systems(rmd_b)
+        )
+
+        return heat_rejection_loop_ids_b
+
+    def create_data(self, context, data):
+        rmd_b = context.baseline
+        heat_rejection_loop_ids_b = (
+            get_heat_rejection_loops_connected_to_baseline_systems(rmd_b)
+        )
+
+        number_of_baseline_heat_rejections_b = sum(
+            [
+                1
+                for heat_rejection_b in find_all("$.heat_rejections[*]", rmd_b)
+                if getattr_(heat_rejection_b, "heat_rejections", "loop")
+                in heat_rejection_loop_ids_b
+            ]
+        )
```

Comment:
Another way to do it could be:

```python
number_of_baseline_heat_rejections_b = len(
            [
                heat_rejection_b
                for heat_rejection_b in find_all("$.heat_rejections[*]", rmd_b)
                if getattr_(heat_rejection_b, "heat_rejections", "loop")
                in heat_rejection_loop_ids_b
            ]
        )
```

Reply:
Good point! Addressed.